src/lib.rs

```diff
@@ -184,8 +183,8 @@ impl Bmp280Builder {
 
     /// Set the path of the I2C device for the sensor.  There is a default value for this, so you
     /// do not need to specify it explicitly.
-    pub fn path(&mut self, path: String) -> &mut Self {
-        self.i2c_path = path;
+    pub fn path(&mut self, path: &str) -> &mut Self {
```

Comment:
Just to check, are you suggesting a change like 
```rust
pub fn path(&mut self, path: PathBuf) -> &mut Self {
    self.i2c_path = path;
    self
}
// Case 1: means calling the Builder would look like: 
let mut dev = Bmp280Builder::new()
    .path(PathBuf::from("/dev/i2c-1"))
    .build()?; 
```
or
```rust
pub fn path(&mut self, path: &str) -> &mut Self {
    self.i2c_path = PathBuf::from(path);
    self
}
// Case 2: means calling the Builder would look like: 
let mut dev = Bmp280Builder::new()
    .path("/dev/i2c-1")
    .build()?; 
```
I'm not exactly sure how the allocation operation changes between the two, to be honest, but as far as I can tell either version will compile as long as the `Bmp280Builder` struct has the `i2c_path: PathBuf` typed field. I'm personally a fan of Case 2 (for whatever reason, just passing the &str to a Builder struct has always looked cleaner to me), but happy to keep iterating on it if this wasn't what you had in mind.

Reply:
So I was suggesting an API like this

```
pub fn path(&mut self, path: impl Into<PathBuf>) -> &mut Self {
  self.i2c_path = path.into();
  self
}
```

This way, if you already have a `PathBuf`, you can pass it to the constructor and avoid an allocation since `T.into::<T>()` is a no-op, but if you have a `&str` or something, you can still pass that in and it will be converted. It's just a more flexible API design that allows the user to make a choice.

Reply:
With this API, either calling convention that you specified in your comment would be supported.

Reply:
Makes sense! I just made those changes, let me know if you see anything else you'd like me to change before merging.

Reply:
Nope, I think this looks great! I'll go ahead and merge this and release `0.4.0` right now